eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs

```diff
@@ -291,14 +311,19 @@ decodeCostModelMap = decodeMapByKey fromCBOR decodeCostModel
 
 decodeCostModel :: Language -> Decoder s CostModel
 decodeCostModel lang = do
+  let keys = costModelParamsNamesSet lang
```

Comment:
I'll make a follow up issues for this, but as a mentioned above, we can now change the "visible" cost model from `Map Text Integer` to `[Integer]`. Which will make this decoder even simpler. :) I suspect we'll have no need for the `costModelParamsNames` anywhere.